packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -220,8 +220,11 @@ const getDefaultEnv = (context: $TSContext): string | undefined => {
       resolution: INVALID_ENV_NAME_MSG,
     });
   }
-
-  if (isNewProject(context) || !context.amplify.getAllEnvs().includes(defaultEnv)) {
+  if (
+    isNewProject(context) ||
+    !context.amplify.getAllEnvs().includes(defaultEnv) ||
+    (context.exeInfo.inputParams && context.exeInfo.inputParams.yes)
```

Comment:
This condition was added so the error message would return the default environment name of `dev` instead of `undefined`